File: regexlib/2021-5-15/javascript_test_file/regexlib_8480.js
/* 8480
 * (http://|)(www\.)?([^\.]+)\.(\w{2}|(com|net|org|edu|int|mil|gov|arpa|biz|aero|name|coop|info|pro|museum))$
 * POLYNOMIAL
 * nums:3
 * POLYNOMIAL AttackString:""+"http://"*5000+"! _1_POA(i)"
 */
var REGEX = new RegExp("(http://|)(www\.)?([^\.]+)\.(\w{2}|(com|net|org|edu|int|mil|gov|arpa|biz|aero|name|coop|info|pro|museum))$")
for(var i = 1; i <= 5000000; i++) {
    var time = Date.now();
    var attack_str = '' + 'http://'.repeat(i*10000) + '! _1_POA(i)'
    REGEX.exec(attack_str)
    // REGEX.test(attack_str);
    // attack_str.search(REGEX)
    // attack_str.match(REGEX)
    var time_cost = Date.now() - time;
    console.log(i * 10000 + ": " + time_cost+" ms")
}

File: regexlib/2021-5-15/python_re_test_file/regexlib_8480.py
# 8480
# (http://|)(www\.)?([^\.]+)\.(\w{2}|(com|net|org|edu|int|mil|gov|arpa|biz|aero|name|coop|info|pro|museum))$
# POLYNOMIAL
# nums:4
# POLYNOMIAL AttackString:""+"http://"*5000+"! _1_POA(i)"

import re
from time import perf_counter

regex = """(http://|)(www\.)?([^\.]+)\.(\w{2}|(com|net|org|edu|int|mil|gov|arpa|biz|aero|name|coop|info|pro|museum))$"""
REGEX = re.compile(regex)
for i in range(0, 150000):
    ATTACK = "" + "http://" * i * 10000 + "! _1_POA(i)"
    LEN = len(ATTACK)
    BEGIN = perf_counter()
    m = REGEX.search(ATTACK)
    # m = REGEX.match(ATTACK)
    DURATION = perf_counter() - BEGIN
    print(f"{i *10000}: took {DURATION} seconds!")